spec/classes/repo_spec.rb

```diff
@@ -3,24 +3,50 @@
 describe 'candlepin::repo' do
   on_supported_os.each do |os, os_facts|
     context "on #{os}" do
-      let(:facts) { os_facts }
-      let(:params) do
-        { version: 'nightly' }
-      end
+      describe 'with nightly version' do
+        let(:facts) { os_facts }
```

Comment:
You can keep the facts in the shared context:
```suggestion
      let(:facts) { os_facts }

      describe 'with nightly version' do
```

The way to think about is it that `context` sets a, well, context. Then everything in that will inherit it. You can further refine the context as well with further nesting.